### PR TITLE
Initial dwarf commit (Sprint 1)

### PR DESCRIPTION
**Pull request by Carbonhell**
PR ready

Implemented features:
-) Dwarf species, they breathe co2 and only speak dwarven
-) Dwarven language
-) Alcohol requirement, calculated based on amount and power of the chemical assumed
-) Small fix to add a hands offset for species (doesn't seems to be required for the current dwarf icon, but said icon causes some issues and might need to be changed)
-) Dwarven tongue and lungs (for species characteristics)

PR description:
Adds the dwarf species. They're similar to humans regarding body features, they have their own sprites obviously. They have special mutant lungs and tongue:
1) Lungs are different because this species relies on CO2 to live. Oxygen isn't toxic but is pointless, without CO2 they suffocate.
2) The tongue restricts them to the dwarven language.
They also spawn with a beard, randomly chosen between Dwarf Beard, Very Long Beard and Full Beard.
They have special behaviour implemented when they assume alcohol: any kind of alcohol (/ethanol child) simply increases their dwarfDrunkness value (100 is good, 0 will probably lead you to death). They lose one unit every 10-15 seconds.
The alcohol system works based in ranges of this variable:
0 to 30: 10 brute loss, 80 stamina loss and a message pointing out you're actually dying if you manage to not notice by the fact you walk like a snail and your flesh's falling off.
30 to 45: danger message, 5% chance to gain a mild brain trauma, 60 stamina damage
45 to 60: 30% chance to print a danger message, 40 stamina loss
60 to 75: 30% chance for the message

The language is based on dwarven words picked off https://dwarffortresswiki.org/index.php/Language and http://www.bay12forums.com/smf/index.php?topic=173289 . They are random words, mostly grammar ones and some nouns thrown in the midde, and the chance for a space to happen is low (35%), so it'll result in quite big words and phrases made of 3-4 words maximum most of the time.
Lastly, they cannot wear any jumpsuit as for now.
To make dwarven jumpsuits, add this species' path to the clothing's species_exception list.

### CODE REVIEW

**Comment by Carbonhell** [comment]
Offsets fixed, taken from https://github.com/tgstation/tgstation/pull/33147-
Jumpsuits are listed in the no_equip list now, future dwarf only jumpsuits will need to add the dwarf species to their species_exception list to make them wearable.

**Comment by ghost** [comment]
make dwarves rip clothes with their weird proportions if they try to put human clothes on resulting in a "ripped rag" they can wear?

**Comment by FantasticFwoosh** [comment]
> make dwarves rip clothes with their weird proportions if they try to put human clothes on resulting in a "ripped rag" they can wear?

I think you have your statement back to front, they arent tiny but they'd be egible to slip through trouser legs etc. Goof's PR had whitelisting of clothes that fit dwarves perfectly just to demystify whitelisting all clothes types they can possibly wear.

**Comment by ErwinSnowden** [comment]
Add the ability to create Keening, Sunder and Wraithguard as endgame equivalents for the Dwemer, I mean dwarves. They can use it on anomalous crystals from the colossus or some shit because it's the heart of a dead god :-). Also can they take Bacchus' Blessing without fucking dying instantly of liver failure? Might want to make them toximmune or use some sort of workaround.

**Comment by Manly-Dwarf** [comment]
don't request shit here, that goes in the dwarfcord or the main discord
carbonhell is not going to take your requests

**Comment by Carbonhell** [comment]
Fixing the jumpsuit grayed out slot. Once that's done, the PR is ready.

File: hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+			dwarfDrunkness = CLAMP(dwarfDrunkness + boozePower, 0, 100)
+			return TRUE // Don't metabolize alcohol like normal humans do.
+
+/datum/species/dwarf/spec_life(mob/living/carbon/human/H)
+	..()
+	if(notDrunkEnoughTime < world.time)
+		dwarfDrunkness--
+		notDrunkEnoughTime = world.time + DRUNK_ALERT_TIME_OFFSET + rand(0, DRUNK_ALERT_TIME_OFFSET/2) // between 10 and 15 seconds
+		switch(dwarfDrunkness) 
+			if(0 to 30) // too low, harmful
```
Comment (Manly-Dwarf) [inline hippiestation/code/modules/mob/living/carbon/human/species_types/dwarves.dm:55]: >Vivadel

**Comment by Carbonhell** [comment]
PR is ready to be discussed

**Comment by HighPrincessErinys** [comment]
good

**Comment by mkikt4743** [comment]
good PR, but should lose drunkeness slower

**Comment by Manly-Dwarf** [comment]
doesn't seem like the whole thing that actually made the species was included
also brute damage when not drunk is probably one of the worst things i have ever heard fuck you vivadel

**Comment by Carbonhell** [comment]
There was a nasty bug related to beards, fixed now

**Comment by Maroonenahmoron** [comment]
Replace critical drunkenness with a brain traumatic tantrum

**Comment by notoriousBIGGIOS** [comment]
add

**Comment by SireTurret** [comment]
remove the "getting damage if you're not drunk" part

it's cool but it'll make dwarves very painful to play

**Comment by TrickyDickRichard** [comment]
> remove the "getting damage if you're not drunk" part
> 
> it's cool but it'll make dwarves very painful to play

yea bro just make it so theyre slower and they work slower without alcohol

**Comment by ghost** [comment]
make an universal work speed factor like in rimworld and it's affected by how drunk a dwarf is or how skilled it is. maybe beard length?

**Comment by TrickyDickRichard** [comment]
NOT skills we're NOT doing skills that would be STUPID

**Comment by Carbonhell** [comment]
Ask Vivadel, I'm implementing the details he gives me. I cannot implement stuff without his consent.

**Comment by Manly-Dwarf** [comment]
local man uses money to rewrite dwarves design

**Comment by Manly-Dwarf** [comment]
far better than nothing at least

**Comment by FantasticFwoosh** [comment]
> Ask Vivadel, I'm implementing the details he gives me. I cannot implement stuff without his consent.

There is precedent in the code in relation to tool speed in reducing it when within a certain criteria. Maybe it could be done by putting the dwarf in a 'sober' or suffering #define range versus the opposite.

[like in surgery -linked](https://github.com/tgstation/tgstation/blob/98afb22229806bc609d6463c9c7e8f18c693e9ed/code/modules/surgery/surgery_step.dm#L73), though .toolspeed does appear here and there in bits and places.

**Comment by Carbonhell** [comment]
Soo any news?

**Comment by Manly-Dwarf** [comment]
This is ready, I'll add in some tuning in another PR coming up quickly (i.e the things you LAZED OUT on doing because "i was not paid for that")
Go demand your money from Vivadel.

**Comment by Manly-Dwarf** [comment]
gah wrong fuckin button

**Pull request merged**